src/test-run/index.js

```diff
@@ -261,3 +300,42 @@ ServiceMessages[CLIENT_MESSAGES.ready] = function (msg) {
 
     return new Promise((resolve, reject) => this.pendingRequest = { resolve, reject, responseTimeout });
 };
+
+ServiceMessages[CLIENT_MESSAGES.executePendingBrowserManipulation] = async function (msg) {
+    this.debugLog.driverMessage(msg);
```

Comment:
I think it makes sense to move browser manipulation logic to separate class `BrowserManipulationManager` and store `screenshotCapturer` inside it, instead of `TestRun`. Test run will own `browserManipulationManager` instanse and will call it's methods like `takeScreenshot` and `takeScreenshotOnFail`